code/modules/atmospherics/machinery/components/trinary_devices/hypertorus.dm

```diff
@@ -0,0 +1,893 @@
+
+	var/scaled_m_plasma = max((m_plasma - FUSION_MOLE_THRESHOLD) / scale_factor, 0)
+	var/scaled_m_nitrogen = max((m_nitrogen - FUSION_MOLE_THRESHOLD) / scale_factor, 0)
+	var/scaled_m_co2 = max((m_co2 - FUSION_MOLE_THRESHOLD) / scale_factor, 0)
+	var/scaled_m_h2o = max((m_h2o - FUSION_MOLE_THRESHOLD) / scale_factor, 0)
+	var/scaled_m_freon = max((m_freon - FUSION_MOLE_THRESHOLD) / scale_factor, 0)
+	var/scaled_m_bz = max((m_bz - FUSION_MOLE_THRESHOLD) / scale_factor, 0)
+	var/scaled_m_proto_nitrate = max((m_proto_nitrate - FUSION_MOLE_THRESHOLD) / scale_factor, 0)
+	var/scaled_m_antinoblium = max((m_antinoblium - FUSION_MOLE_THRESHOLD) / scale_factor, 0)
+	var/scaled_m_hypernoblium = max((m_hypernoblium - FUSION_MOLE_THRESHOLD) / scale_factor, 0)
+
+	/*
+	 *FUSION MAIN PROCESS
+	 */
+	//This section is used for the instability calculation for the fusion reaction
+	//The size of the phase space hypertorus
+	var/toroidal_size = (2 * PI) + TORADIANS(arctan((volume - TOROID_VOLUME_BREAKEVEN) / TOROID_VOLUME_BREAKEVEN))
+	//Calculation of the gas power, only for theoretical instability calculations
+	var/gas_power = 0
+	for (var/gas_id in internal_fusion.gases)
+		gas_power += (internal_fusion.gases[gas_id][GAS_META][META_GAS_FUSION_POWER] * internal_fusion.gases[gas_id][MOLES])
+	for (var/gas_id in moderator_internal.gases)
+		gas_power += (moderator_internal.gases[gas_id][GAS_META][META_GAS_FUSION_POWER] * moderator_internal.gases[gas_id][MOLES] * 0.75)
+
+	instability = MODULUS((gas_power * INSTABILITY_GAS_POWER_FACTOR)**2, toroidal_size) + current_damper + iron_content
+	//Effective reaction instability (determines if the energy is used/released)
+	var/internal_instability = 0
+	if(instability * 0.5 < FUSION_INSTABILITY_ENDOTHERMALITY)
+		internal_instability = 1
+	else
+		internal_instability = -1
+
+	/*
+	 *Modifiers
+	 */
+	///Those are the scaled gases that gets consumed and releases energy or help increase that energy
+	var/positive_modifiers = scaled_hydrogen + scaled_tritium + scaled_m_nitrogen * 0.35 + scaled_m_co2 * 0.55 + scaled_m_antinoblium * 10 - scaled_m_hypernoblium * 10
+	///Those are the scaled gases that gets produced and consumes energy or help decrease that energy
+	var/negative_modifiers = scaled_helium + scaled_m_h2o * 0.75 + scaled_m_freon * 1.15 - scaled_m_antinoblium * 10
+	///Between 0.25 and 100, this value is used to modify the behaviour of the internal energy and the core temperature based on the gases present in the mix
+	var/power_modifier = clamp(scaled_tritium * 1.05 + scaled_m_co2 * 0.95 + scaled_m_plasma * 0.05 - scaled_helium * 0.55 - scaled_m_freon * 0.75, 0.25, 100)
+	///Minimum 0.25, this value is used to modify the behaviour of the energy emission based on the gases present in the mix
+	var/heat_modifier = max(scaled_hydrogen * 1.15 + scaled_helium * 1.05 + scaled_m_plasma * 1.25 - scaled_m_nitrogen * 0.75 - scaled_m_freon * 0.95, 0.25)
```

Comment:
Currently heat_modifier has no effect beyond 100. Suggested changes reflect that. If you have absolutely no plan to use heat_modifier past 100, please consider clamping here and simplify `min(heat_modifier, 100)` to `heat_modifier` instead of applying the suggestions.